packages/with_bloc/lib/src/with_bloc.dart

```diff
@@ -1,30 +1,53 @@
 import 'package:collection/collection.dart';
-import 'package:flutter/foundation.dart';
-import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
 
 const listId = ListEquality<dynamic>();
 
+/// A helper widget for easily creating [StateQueue] in the [build] method.
+///
+/// This widget takes care of creating the [BlocType] and disposing it when necessary.
+/// It also handles recreating the [BlocType] whenever any [inputs] change.
 class WithBloc<BlocType extends ValueNotifier<StateType>, StateType>
     extends StatefulWidget {
   const WithBloc({
     Key key,
     @required this.createBloc,
-    this.inputs,
     @required this.builder,
+    this.inputs = const [],
     this.child,
   })  : assert(createBloc != null),
         assert(builder != null),
+        assert(inputs != null),
         super(key: key);
 
+  /// A function which creates the [BlocType] instance
+  ///
+  /// This function might be called multiple times in the lifetime of this widget.
+  /// Whenever the [inputs] change, a new bloc will be created by calling this function.
   final BlocType Function(BuildContext context) createBloc;
 
+  /// A [Function] which builds a widget depending on the [BlocType] and [StateType].
+  ///
+  /// The [bloc] will be the current bloc and [value] is the current [StateType].
+  /// When the bloc emits a new value, this function will be called again to reflect the changes in the UI.
+  ///
+  /// The [child] will be passed in as the fourth argument and might be `null` if not supplied.
   final Widget Function(
     BuildContext context,
     BlocType bloc,
     StateType value,
     Widget child,
   ) builder;
 
+  /// A [BlocType]-independent widget which is passed back to the [builder].
+  ///
+  /// This argument is optional and can be null if the entire widget subtree
+  /// the [builder] builds depends on the value of the [BlocType].
+  ///
+  /// Can be `null`.
```

Comment:
Is this redundant on purpose, or was it left over when switching from our usual nullability comments to the more elaborate one above?

Reply:
Yes, this was duplicate. I first added the `Can be null` and then added the more descriptive text